pkg/syncer/syncer.go

```diff
@@ -19,13 +19,15 @@ limitations under the License.
 package syncer
 
 import (
+	"context"
+
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
 type Interface interface {
 	Start(stopCh <-chan struct{}) error
-	AwaitStopped()
+	AwaitStopped(ctx context.Context) error
 	GetResource(name, namespace string) (runtime.Object, bool, error)
```

Comment:
Not a blocker for this PR, but shouldn’t all the syncer operations accept a context and handle cancellation?

Reply:
Perhaps but this and others are  non-blocking and internal invocations don't take a context.